### PR TITLE
Feature/code refractor

### PR DESCRIPTION
**Pull request by Sean-Chuang**
code refractor

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+@app.route('/msg')
+def msg():
+    text = request.args.get('text', None)
+    channel = request.args.get('channel', None)
+    bot.send_msg(text, channel)
+    return Response(status = 200)
+
+if __name__ == '__main__':
+    app.run(host='0.0.0.0')
```
Comment (yetihan) [inline app.py]: Why not use the config here since you have written  `FLASK_RUN_HOST`  in .env

File: .env [hunk 2]
```diff
@@ -0,0 +1,15 @@
+FLASK_ENV=development
+FLASK_APP=app.py
+FLASK_RUN_HOST=0.0.0.0
+FLASK_RUN_PORT=8080
+
+# System setting
+SYS_IM_TYPE=slack
+
+# Salck
+SLACK_DEFAULT_CHANNEL=#project-gakki
```
Comment (yetihan) [inline .env]: It works without `#` in channel.

File: app.py (hunk 1 above)
Comment (Sean-Chuang) [reply app.py]: Oh, ya, ok

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+python-dotenv
+flask
```
Comment (yetihan) [inline requirements.txt]: Miss some requirements?

File: app.py (hunk 1 above)
Comment (Sean-Chuang) [reply app.py]: But let me explain here. right now we use flask run, we would not use the code here. flask run would auto check `FLASK_RUN_HOST`

File: config.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import os
+from importlib import import_module
+
+class Config(object):
+    SYS_IM_TYPE = os.environ.get("SYS_IM_TYPE")
```
Comment (ZhangQiuLu-SmartNews) [inline config.py:5]: I think you can try GLAGS?
https://github.com/google/python-gflags

Comment (ZhangQiuLu-SmartNews) [inline config.py:5]: I think you can use gflags to handle config?
https://github.com/google/python-gflags

Comment (Sean-Chuang) [reply config.py:5]: Maybe next time. I think It is just  a simple config. Let user check by themselves.  If wrong, then server can't start.

File: app.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+from config import Config
+from logger import logger
+from flask import Flask
```
Comment (ZhangQiuLu-SmartNews) [inline app.py:3]: using ,?

File: config.py (hunk 4 above)
Comment (ZhangQiuLu-SmartNews) [reply config.py:5]: haha yeap~

File: app.py (hunk 5 above)
Comment (Sean-Chuang) [reply app.py:3]: Oh, I guess it is both ok.
https://stackoverflow.com/questions/15011367/python-module-import-single-line-vs-multi-line
There also have some argue about this. 
 From a stylistic perspective, one might be more preferable than the other.

**Review by ZhangQiuLu-SmartNews — APPROVED** [review]
LGTM

**Review by yetihan — APPROVED** [review]
LGTM

File: .env (hunk 2 above)
Comment (Sean-Chuang) [reply .env]: really,  let me have. a try.

File: requirements.txt (hunk 3 above)
Comment (Sean-Chuang) [reply requirements.txt]: oh, slack.😱

**Pull request merged**